test/e2e/antreapolicy_test.go

```diff
@@ -655,50 +659,49 @@ func testANPBasic(t *testing.T) {
 	builder = builder.SetName("test-log-cnp-deny").
 		SetPriority(1.0).
-		SetAppliedToGroup(map[string]string{"pod": "a"}, nil, nil, nil)
+		SetAppliedToGroup(map[string]string{"pod": "a"}, map[string]string{"ns": "x"}, nil, nil)
 	builder.AddEgress(v1.ProtocolTCP, &p80, nil, nil, nil, map[string]string{"ns": "z"},
-		nil, nil, secv1alpha1.RuleActionDrop).
-		AddEgressLogging()
+		nil, nil, secv1alpha1.RuleActionDrop)
+	builder.AddEgressLogging()
 
-	reachability := NewReachability(allPods, true)
-	reachability.Expect(Pod("x/a"), Pod("z/a"), false)
-	reachability.Expect(Pod("x/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("x/a"), Pod("z/c"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/a"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("y/a"), Pod("z/c"), false)
-	reachability.Expect(Pod("z/a"), Pod("z/b"), false)
-	reachability.Expect(Pod("z/a"), Pod("z/c"), false)
+	_, err := k8sUtils.CreateOrUpdateCNP(builder.Get())
+	failOnError(err, t)
+	time.Sleep(networkPolicyDelay)
 
-	testStep := []*TestStep{
-		{
-			"Port 80",
-			reachability,
-			[]metav1.Object{builder.Get()},
-			80,
-			0,
-		},
+	// generate some traffic that will be dropped by test-log-cnp-deny
+	k8sUtils.Probe("x", "a", "z", "a", p80)
+	k8sUtils.Probe("x", "a", "z", "b", p80)
+	k8sUtils.Probe("x", "a", "z", "c", p80)
+	time.Sleep(networkPolicyDelay)
+
+	podXA, _ := k8sUtils.GetPod("x", "a")
+	// nodeName is guaranteed to be set at this stage, since the framework waits for all Pods to be in Running phase
+	nodeName := podXA.Spec.NodeName
+	antreaPodName, err := data.getAntreaPodOnNode(nodeName)
+	if err != nil {
+		t.Errorf("error occurred when trying to get the Antrea Agent pod running on node %s: %v", nodeName, err)
 	}
-	testCase := []*TestCase{
-		{"Audit Log CNP Drop Egress From All Pod:a to NS:z", testStep},
+	cmd := []string{"cat", logDir + logfileName}
+	stdout, stderr, err := data.runCommandFromPod(antreaNamespace, antreaPodName, "antrea-agent", cmd)
+	if err != nil || stderr != "" {
+		t.Errorf("error occurred when inspecting the audit log file. err: %v, stderr: %v", err, stderr)
 	}
-	executeTests(t, testCase)
-}
+	assert.Equalf(t, true, strings.Contains(stdout, "test-log-cnp-deny"), "audit log does not contain entries for test-log-cnp-deny")
 
-// validateAuditLog validates whether testAuditLoggingBasic succeeded or failed
-func validateAuditLog() bool {
-	data, err := ioutil.ReadFile("/var/log/antrea/networkpolicy/np.log")
-	if err != nil {
-		return false
+	destinations := []string{"z/a", "z/b", "z/c"}
+	srcIP, _ := podIPs["x/a"]
+	for _, d := range destinations {
+		dstIP, _ := podIPs[d]
+		// The audit log should contain log entry `... Drop <ofPriority> SRC: <x/a IP> DEST: <z/* IP> ...`
+		pattern := `Drop [0-9]+ SRC: ` + srcIP + ` DEST: ` + dstIP
+		match, _ := regexp.MatchString(pattern, stdout)
+		assert.Equalf(t, true, match, "audit log does not contain expected entry for x/a to %s", d)
```

Comment:
Use `assert.Regexp` instead to save lines?

Reply:
Thanks for the suggestion. Done